pkg/ccl/changefeedccl/kafkaauth/kafkaauth.go

```diff
@@ -0,0 +1,181 @@
+// Copyright 2025 The Cockroach Authors.
+//
+// Use of this software is governed by the CockroachDB Software License
+// included in the /LICENSE file.
+
+package kafkaauth
+
+import (
+	"context"
+	"sort"
+	"strings"
+
+	"github.com/IBM/sarama"
+	"github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/changefeedbase"
+	"github.com/cockroachdb/errors"
+	"github.com/twmb/franz-go/pkg/kgo"
+)
+
+type saslMechanismBuilder interface {
+	name() string
+	validateParams(u *changefeedbase.SinkURL) error
```

Comment:
maybe this should be `requiredParams([]string)`, since that's what all the mechanisms do